pkg/storage/tsdb/labels.go

```diff
@@ -23,3 +26,48 @@ func FromLabelAdaptersToLabels(input []client.LabelAdapter) labels.Labels {
 
 	return result
 }
+
+// FromLabelsToLabelAdapters converts TSDB labels.labels to []LabelAdapter.
```

Comment:
Is all this duplication of labels and conversions necessary? :-(

Reply:
Thanks for questioning it. I re-checked it, and it should be safe to do a data type conversion only here.

What's about now?

Reply:
I was questioning the need to have labels.Labels vs client.LabelAdapter vs legacy_labels.Labels... they all seem pretty much the same to me. I don't like it, but I assume there is a good reason :-(

Reply:
As of today, we need it in order to reuse the large part of Cortex which works with `client.LabelAdapter`. At least now we just do a data type conversion.